File: ovvium-backend/src/main/java/com/ovvium/services/web/controller/bff/v1/transfer/response/payment/UserCardDataResponse.java
package com.ovvium.services.web.controller.bff.v1.transfer.response.payment;

import com.ovvium.services.repository.client.payment.dto.InfoUserResponse;
import lombok.Data;

import java.util.UUID;

@Data
public final class UserCardDataResponse {

	private final UUID pciDetailsId;
	private final String pan;
	private final String brand;
	private final String type;
	private final String country;
	private final String expiryDate;
	private final String hash;
	private final String category;
	private final Integer sepa;

	public UserCardDataResponse(UUID pciDetailsId, InfoUserResponse response) {
		this.pciDetailsId = pciDetailsId;
		this.pan = response.getCardPan();
		this.brand = response.getCardBrand();
		this.type = response.getCardType();
		this.country = response.getCardCountry();
		this.expiryDate = response.getCardExpiryDate();
		this.hash = response.getCardHash();
		this.category = response.getCardCategory();
		this.sepa = response.getCardSepa();
	}

}

File: ovvium-app/src/model/response/UserCardDataResponse.ts
export class UserCardDataResponse {

	pciDetailsId: string;
	pan: string;
	brand: string;
	type: string;
	country: string;
	expiryDate: string;
	hash: string;
	category: string;
	sepa: string;

	constructor(response: UserCardDataResponse) {
		this.pciDetailsId = response.pciDetailsId;
		this.pan = response.pan;
		this.brand = response.brand;
		this.type = response.type;
		this.country = response.country;
		this.expiryDate = response.expiryDate;
		this.hash = response.hash;
		this.category = response.category;
		this.sepa = response.sepa;
	}

}
